pkg/graph/resolvers/trb_request_form.go

```diff
@@ -5,14 +5,22 @@ import (
 	"errors"
 
 	"github.com/google/uuid"
+	"golang.org/x/sync/errgroup"
 
 	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/email"
 	"github.com/cmsgov/easi-app/pkg/models"
 	"github.com/cmsgov/easi-app/pkg/storage"
 )
 
 // UpdateTRBRequestForm updates a TRBRequestForm record in the database
-func UpdateTRBRequestForm(ctx context.Context, store *storage.Store, input map[string]interface{}) (*models.TRBRequestForm, error) {
+func UpdateTRBRequestForm(
+	ctx context.Context,
+	store *storage.Store,
+	emailClient *email.Client,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
```

Comment:
@DylanSpOddball, I wonder here if it might be better to provide the UserInfo to the resolver before this endpoint, instead of providing a delegate to retrieve it?  I think this works, but I wonder if it's mixing responsibilities?  What do you think?

I was having trouble testing this locally, but I realized it's because of the new paradigm for MOCK user data. I was  using an EUAID not related to one of the test accounts. It worked when I switched to one that did have a relevant mock account.

Reply:
I wasn't entirely sure about how to structure this. I think it's the resolver's general responsibility to aggregate data like this, and the convention we've been following is to put all logic into the files under `pkg/graph/resolvers/` instead of in `schema.resolvers.go`. The reason to pass a delegate in is to pass the function that's assigned to `r.service.FetchUserInfo`; we can't directly call the `cedarldap.Client`'s `FetchUserInfo()` method, because we need an instance of `cedarldap.Client` to call it on. I suppose we could add a field on the `Resolver` struct to store a `cedarldap.Client` instance, pass that in from `schema.resolvers.go` to this resolver function, then call its `FetchUserInfo()` method, but that doesn't seem like it'd make much of a difference.

Reply:
I think you are right, I think this is a fine way to do it. When we've done something similar in MINT, we've just got the user information by calling the endpoint in the schema.resolvers.go, then passed the userInfo struct along. 

I kind of like that because then the resolver doesn't really need to handle getting any user info, it just knows to expect that the user info has fetched. But, I think you could argue what you've done here with the delegate is basically that, but perhaps more elegant. Thanks for sharing your thoughts!

Reply:
And I also just realized another implication, you don't fetch the info unless you actually need it this way, which is another bonus!